status-counter.js

```diff
@@ -12,66 +15,53 @@ class CounterWrap extends Wrap {
    * cassandra driver until we put this into datastar in a proper way
    *
    * @function increment
-   * @param {Object} params Parameters for performing increment
-   * @param {String} params.env Env for Counter table
-   * @param {String} params.pkg Pkg for Counter table
-   * @param {String} params.version Version for counter table
+   * @param {Object} key Object containing key of item to increment
    * @param {Number} amount Number to increment
    * @returns {Promise} when operation complete
    * @public
    */
-  async increment(params, amount = 1) {
-    const client = await this._getConnection();
-    const query = `UPDATE status_counter SET count=count+${amount} WHERE pkg=:pkg AND env=:env AND version=:version`;
-    return client.execute(query, params, { prepare: true, counter: true });
+  async increment(key, amount = 1) {
+    return this.update({ ...key, count: { $add: amount } });
   }
   /**
    * Decrement the given counter for this table in the manual way with the
    * cassandra driver until we put this into datastar in a proper way
    *
    * @function increment
-   * @param {Object} params Parameters for performing increment
-   * @param {String} params.env Env for Counter table
-   * @param {String} params.pkg Pkg for Counter table
-   * @param {String} params.version Version for counter table
+   * @param {Object} key Object containing key of item to increment
    * @param {Number} amount Number to increment
    * @returns {Promise} when operation complete
    * @public
    */
-  async decrement(params, amount = 1) {
-    const client = await this._getConnection();
-    const query = `UPDATE status_counter SET count=count-${amount} WHERE pkg=:pkg AND env=:env AND version=:version`;
-    return client.execute(query, params, { prepare: true, counter: true });
+  async decrement(key, amount = 1) {
+    return this.update({ ...key, count: { $add: -1 * amount } });
```

Comment:
ah nice so this does pass through correctly to our underlying DynamoDB driver without us exposing anything?